### PR TITLE
New tests and hardening of actions updating

### PR DESCRIPTION
**Pull request by MikeRalphson**
Hi!

Thanks for this library. Some tests ported from my own incomplete implementation (which was based on an even earlier version of the spec).

Let me know if you want me to break this up.

Also I noticed the `package.json` name is `preparse` and there is no `description`.

### CODE REVIEW

File: src/overlay.js [hunk 1]
```diff
@@ -19,34 +19,38 @@ function applyOverlayToOpenAPI(spec, overlay) {
...
-			});
 		}
 	})
 
 	return spec;
 }
 
 function sortOpenAPIFields(field1, field2) {
-    const orderedKeys = ["info", "servers", "paths", "components", "summary", "objectId", "description", "tags", "parameters", "responses"];
+	const orderedKeys = ["info", "servers", "summary", "operationId", "tags", "paths", "components", "description", "parameters", "responses"];
```
Comment (lornajane) [inline src/overlay.js:42]: Thanks, I was hoping someone had thought about this more than I had!

File: test/overlays/overlay.yaml [hunk 2]
```diff
@@ -11,7 +11,7 @@ actions:
       - wildcard-done
 - target: $.paths['/pets'].get
   update:
-    info:
+    x-info:
```
Comment (lornajane) [inline test/overlays/overlay.yaml:14]: Is this change because the info tag doesn't belong there in an operation?

**Comment by lornajane** [comment]
Thanks so much for adding these! I need to have another look to digest them properly but I dropped a couple of comments from just looking at the changes.

File: test/overlays/overlay.yaml (hunk 2 above)
Comment (MikeRalphson) [reply test/overlays/overlay.yaml:14]: Yup, I didn't know if validity of the test case output was worth making the change for... Maybe we could swap it for an `externalDocs`object or something else which does apply there?

File: src/overlay.js (hunk 1 above)
Comment (MikeRalphson) [reply src/overlay.js:42]: It's a bit awkward because tags can appear more than once but there's only one sort precedence for it...

**Review by lornajane — APPROVED** [review]
Thank you! More tests are always SO WELCOME, and you fixed some other very valuable things too. Much appreciated :)

**Pull request merged**